sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/test/environment/setup/AzureSearchResources.java

```diff
@@ -114,13 +115,21 @@ public void createService(TestResourceNamer testResourceNamer) {
         System.out.println("Creating Azure Cognitive Search service: " + searchServiceName);
         int recreateCount = 0;
         do {
-            searchService = azure.searchServices()
-                .define(searchServiceName)
-                .withRegion(location)
-                .withExistingResourceGroup(resourceGroup)
-                .withFreeSku()
-                .create();
-            recreateCount += 1;
+            try {
+                searchService = azure.searchServices()
```

Comment:
We should add some delay between each retry instead of retrying with no delay.

Reply:
There is delay of 30 seconds on every .create() operation.
Put it in shouldRetry() helper.